clippy_lints/src/time_subtraction.rs

```diff
@@ -88,7 +90,8 @@ impl LateLintPass<'_> for InstantSubtraction {
             rhs,
         ) = expr.kind
             && let typeck = cx.typeck_results()
-            && ty::is_type_diagnostic_item(cx, typeck.expr_ty(lhs), sym::Instant)
+            && (ty::is_type_diagnostic_item(cx, typeck.expr_ty(lhs), sym::Instant)
+                || ty::is_type_diagnostic_item(cx, typeck.expr_ty(lhs), sym::Duration))
```

Comment:
You can remove one lookup with:

```suggestion
            && matches!(
                ty::get_type_diagnostic_name(cx, typeck.expr_ty(lhs)),
                Some(sym::Instant | sym::Duration)
            )
```